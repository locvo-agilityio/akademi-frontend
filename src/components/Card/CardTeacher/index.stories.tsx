import { StoryObj, Meta } from '@storybook/react';

import CardTeacher from '.';

const meta = {
  title: 'Components/Cards/CardTeacher',
  component: CardTeacher,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
} satisfies Meta<typeof CardTeacher>;

export default meta;

type Story = StoryObj<typeof CardTeacher>;

export const Default: Story = {
  args: {
    name: 'John Doe',
    subject: 'Math Teacher',
    avatar: 'https://bit.ly/dan-abramov',
    phone: '123-456-7890',
    email: 'K5K3x@example.com',
  },
};
