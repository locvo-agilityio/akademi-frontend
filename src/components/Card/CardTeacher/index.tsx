import { memo, MouseEvent, useCallback } from 'react';
import {
  Avatar,
  Button,
  Card,
  CardBody,
  CardHeader,
  Flex,
  Icon,
  Text,
  Tooltip,
  VStack,
} from '@chakra-ui/react';
import { useNavigate } from 'react-router-dom';

// Icons
import { MailIcon, PhoneIcon } from '@/components/icons';

// Components
import Dropdown from '@/components/Dropdown';

// Constants
import { PUBLIC_ROUTERS } from '@/constants';

interface CardTeacherProps {
  documentId: string;
  name: string;
  subject: string;
  avatar?: string;
  phone: string;
  email: string;
  onOpenDeleteModal?: (e: MouseEvent<HTMLElement>) => void;
}

const CardTeacher = ({
  documentId,
  name,
  subject,
  avatar = '',
  phone,
  email,
  onOpenDeleteModal,
}: CardTeacherProps) => {
  const navigate = useNavigate();

  const handleNavigateToTeacherDetail = useCallback(() => {
    navigate(PUBLIC_ROUTERS.TEACHER_DETAIL.replace(':id', documentId));
  }, [documentId, navigate]);

  return (
    <Card
      p={8}
      borderRadius="2xl"
      _hover={{ boxShadow: '2xl' }}
      cursor="pointer"
      onClick={handleNavigateToTeacherDetail}
    >
      <CardHeader
        display="flex"
        justifyContent="space-between"
        gap={6}
        pl="84px"
      >
        <VStack w="full">
          <Avatar size="2xl" name={name} src={avatar} />
        </VStack>
        <Dropdown
          documentId={documentId}
          name={name}
          path={PUBLIC_ROUTERS.TEACHER_DETAIL.replace(':id', documentId)}
          editPath={PUBLIC_ROUTERS.TEACHER_EDIT.replace(':id', documentId)}
          onOpenDeleteModal={onOpenDeleteModal}
        />
      </CardHeader>

      <CardBody
        display="flex"
        flexDirection="column"
        justifyContent="center"
        alignItems="center"
        gap={4}
      >
        <Text as="p" mt={6} fontSize="lg" fontWeight="bold" color="darkblue">
          {name}
        </Text>
        <Text mt={4} as="span" fontSize="md" color="gray.600">
          {subject}
        </Text>

        <Flex gap={4}>
          <Tooltip label={phone}>
            <Button
              as="a"
              aria-label="Call"
              href={`tel:${phone}`}
              variant="icon"
              w={10}
              bgColor="primary"
              color="white"
            >
              <Icon as={PhoneIcon} boxSize={5} />
            </Button>
          </Tooltip>

          <Tooltip label={email}>
            <Button
              as="a"
              aria-label="Email"
              href={`mailto:${email}`}
              variant="icon"
              w={10}
              bgColor="primary"
              color="white"
            >
              <Icon as={MailIcon} boxSize={5} />
            </Button>
          </Tooltip>
        </Flex>
      </CardBody>
    </Card>
  );
};

export default memo(CardTeacher);
