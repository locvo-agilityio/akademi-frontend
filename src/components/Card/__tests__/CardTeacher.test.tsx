import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import CardTeacher from '../CardTeacher';

describe('CardTeacher component', () => {
  const mockProps = {
    id: 1,
    documentId: '1',
    name: 'Dimitres Viga',
    email: 'K5K3x@example.com',
    phone: '123-456-7890',
    subject: 'Mathematics',
    address: 'Jakarta, Indonesia',
  };

  it('should render correctly with default props', () => {
    const { container } = render(
      <BrowserRouter>
        <CardTeacher {...mockProps} />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
