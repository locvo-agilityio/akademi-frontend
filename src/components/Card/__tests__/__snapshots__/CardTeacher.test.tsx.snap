// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTeacher component should render correctly with default props 1`] = `
<div>
  <div
    class="chakra-card css-1sb8yah"
  >
    <div
      class="chakra-card__header css-xon872"
    >
      <div
        class="chakra-stack css-1y8u3mh"
      >
        <span
          class="chakra-avatar css-1ruhx4w"
        >
          <div
            aria-label="Dimitres Viga"
            class="chakra-avatar__initials css-0"
            role="img"
          >
            DV
          </div>
        </span>
      </div>
      <button
        aria-controls="menu-list-:r1:"
        aria-expanded="false"
        aria-haspopup="menu"
        class="chakra-button chakra-menu__menu-button css-12tg24h"
        id="menu-button-:r1:"
        title="Dropdown"
        type="button"
      >
        <span
          class="css-xl71ch"
        />
        <span
          class="chakra-button__icon css-1hzyiq5"
        >
          <svg
            aria-hidden="true"
            fill="none"
            focusable="false"
            height="6"
            viewBox="0 0 24 6"
            width="24"
          >
            <path
              d="M12.0012 0.360107C11.6543 0.360107 11.3109 0.428424 10.9904 0.561157C10.67 0.693889 10.3788 0.888439 10.1335 1.1337C9.88829 1.37896 9.69374 1.67012 9.56101 1.99056C9.42828 2.31101 9.35996 2.65446 9.35996 3.00131C9.35996 3.34816 9.42828 3.69161 9.56101 4.01205C9.69374 4.3325 9.88829 4.62366 10.1335 4.86892C10.3788 5.11418 10.67 5.30873 10.9904 5.44146C11.3109 5.57419 11.6543 5.64251 12.0012 5.64251C12.7017 5.64235 13.3734 5.36393 13.8686 4.86849C14.3638 4.37306 14.6419 3.7012 14.6418 3.00071C14.6416 2.30022 14.3632 1.62848 13.8677 1.13327C13.3723 0.638064 12.7004 0.359948 12 0.360107H12.0012ZM3.60116 0.360107C3.25431 0.360107 2.91086 0.428424 2.59042 0.561157C2.26997 0.693889 1.97881 0.888439 1.73355 1.1337C1.48829 1.37896 1.29374 1.67012 1.16101 1.99056C1.02828 2.31101 0.959961 2.65446 0.959961 3.00131C0.959961 3.34816 1.02828 3.69161 1.16101 4.01205C1.29374 4.3325 1.48829 4.62366 1.73355 4.86892C1.97881 5.11418 2.26997 5.30873 2.59042 5.44146C2.91086 5.57419 3.25431 5.64251 3.60116 5.64251C4.30165 5.64235 4.97339 5.36393 5.4686 4.86849C5.9638 4.37306 6.24192 3.7012 6.24176 3.00071C6.2416 2.30022 5.96318 1.62848 5.46775 1.13327C4.97231 0.638064 4.30045 0.359948 3.59996 0.360107H3.60116ZM20.4012 0.360107C20.0543 0.360107 19.7109 0.428424 19.3904 0.561157C19.07 0.693889 18.7788 0.888439 18.5336 1.1337C18.2883 1.37896 18.0937 1.67012 17.961 1.99056C17.8283 2.31101 17.76 2.65446 17.76 3.00131C17.76 3.34816 17.8283 3.69161 17.961 4.01205C18.0937 4.3325 18.2883 4.62366 18.5336 4.86892C18.7788 5.11418 19.07 5.30873 19.3904 5.44146C19.7109 5.57419 20.0543 5.64251 20.4012 5.64251C21.1017 5.64235 21.7734 5.36393 22.2686 4.86849C22.7638 4.37306 23.0419 3.7012 23.0418 3.00071C23.0416 2.30022 22.7632 1.62848 22.2677 1.13327C21.7723 0.638064 21.1005 0.359948 20.4 0.360107H20.4012Z"
              fill="#A098AE"
            />
          </svg>
        </span>
      </button>
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-abgd6"
          id="menu-list-:r1:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <a
            class="chakra-menu__menuitem css-1dzc0uv"
            data-index="0"
            href="/teachers/edit/1"
            id="menu-list-:r1:-menuitem-:r2:"
            role="menuitem"
            tabindex="-1"
          >
            Edit
          </a>
          <button
            class="chakra-menu__menuitem css-1f187mo"
            data-id="1"
            data-index="1"
            data-name="Dimitres Viga"
            id="menu-list-:r1:-menuitem-:r3:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
    <div
      class="chakra-card__body css-w98boq"
    >
      <p
        class="chakra-text css-2pz5xn"
      >
        Dimitres Viga
      </p>
      <span
        class="chakra-text css-16tsojh"
      >
        Mathematics
      </span>
      <div
        class="css-1utx3w7"
      >
        <a
          aria-label="Call"
          class="chakra-button css-1aix1gy"
          href="tel:123-456-7890"
        >
          <svg
            class="chakra-icon css-1vp16hi"
            fill="none"
            focusable="false"
            height="20"
            viewBox="0 0 21 20"
            width="21"
          >
            <path
              d="M18.4401 11.0001C18.2201 11.0001 17.9901 10.9301 17.7701 10.8801C17.3246 10.7819 16.8868 10.6516 16.4601 10.4901C15.9962 10.3213 15.4862 10.3301 15.0284 10.5147C14.5706 10.6993 14.1972 11.0467 13.9801 11.4901L13.7601 11.9401C12.7861 11.3983 11.8911 10.7253 11.1001 9.94011C10.3149 9.14913 9.64192 8.25411 9.1001 7.28011L9.5201 7.00011C9.96347 6.78303 10.3109 6.40964 10.4955 5.9518C10.6801 5.49396 10.6889 4.98402 10.5201 4.52011C10.3613 4.09253 10.231 3.65491 10.1301 3.21011C10.0801 2.99011 10.0401 2.76011 10.0101 2.53011C9.88867 1.82573 9.51973 1.18785 8.96972 0.731352C8.41972 0.274849 7.7248 0.0297211 7.0101 0.0401122H4.0101C3.57913 0.0360657 3.15235 0.124926 2.75881 0.300643C2.36527 0.47636 2.01421 0.734808 1.72953 1.05839C1.44485 1.38198 1.23324 1.76311 1.10909 2.17583C0.984942 2.58855 0.95118 3.02317 1.0101 3.45011C1.54284 7.63949 3.45613 11.532 6.44775 14.5127C9.43938 17.4935 13.3388 19.3926 17.5301 19.9101H17.9101C18.6475 19.9112 19.3595 19.6406 19.9101 19.1501C20.2265 18.8672 20.4792 18.5203 20.6516 18.1324C20.8239 17.7446 20.9121 17.3245 20.9101 16.9001V13.9001C20.8979 13.2055 20.6449 12.5367 20.1944 12.0078C19.744 11.4789 19.1239 11.1227 18.4401 11.0001ZM18.9401 17.0001C18.9399 17.1421 18.9095 17.2824 18.8509 17.4117C18.7923 17.5411 18.7068 17.6564 18.6001 17.7501C18.4884 17.8466 18.3577 17.9187 18.2165 17.9617C18.0753 18.0048 17.9267 18.0178 17.7801 18.0001C14.035 17.5199 10.5563 15.8066 7.89282 13.1304C5.2293 10.4542 3.53251 6.96745 3.0701 3.22011C3.05419 3.07363 3.06813 2.92544 3.1111 2.7845C3.15407 2.64356 3.22517 2.5128 3.3201 2.40011C3.41381 2.29344 3.52916 2.20795 3.65848 2.14933C3.7878 2.09071 3.92812 2.06029 4.0701 2.06011H7.0701C7.30265 2.05494 7.52972 2.13099 7.71224 2.27518C7.89476 2.41937 8.02131 2.62268 8.0701 2.85011C8.1101 3.12345 8.1601 3.39345 8.2201 3.66011C8.33562 4.18726 8.48936 4.70529 8.6801 5.21011L7.2801 5.86011C7.1604 5.91503 7.05272 5.99306 6.96326 6.08971C6.87379 6.18636 6.8043 6.29973 6.75877 6.42331C6.71324 6.54689 6.69257 6.67824 6.69795 6.80983C6.70332 6.94142 6.73464 7.07066 6.7901 7.19011C8.2293 10.2729 10.7073 12.7509 13.7901 14.1901C14.0336 14.2901 14.3066 14.2901 14.5501 14.1901C14.6748 14.1455 14.7894 14.0766 14.8873 13.9873C14.9851 13.898 15.0643 13.7902 15.1201 13.6701L15.7401 12.2701C16.2571 12.455 16.7847 12.6086 17.3201 12.7301C17.5868 12.7901 17.8568 12.8401 18.1301 12.8801C18.3575 12.9289 18.5608 13.0554 18.705 13.238C18.8492 13.4205 18.9253 13.6476 18.9201 13.8801L18.9401 17.0001Z"
              fill="currentColor"
            />
          </svg>
        </a>
        <a
          aria-label="Email"
          class="chakra-button css-1aix1gy"
          href="mailto:K5K3x@example.com"
        >
          <svg
            class="chakra-icon css-1vp16hi"
            fill="none"
            focusable="false"
            height="16"
            viewBox="0 0 22 16"
            width="22"
          >
            <path
              d="M19 0H3C2.20435 0 1.44129 0.316071 0.87868 0.87868C0.31607 1.44129 0 2.20435 0 3V13C0 13.7956 0.31607 14.5587 0.87868 15.1213C1.44129 15.6839 2.20435 16 3 16H19C19.7956 16 20.5587 15.6839 21.1213 15.1213C21.6839 14.5587 22 13.7956 22 13V3C22 2.20435 21.6839 1.44129 21.1213 0.87868C20.5587 0.316071 19.7956 0 19 0ZM20 12.75L15.1 8.35L20 4.92V12.75ZM2 4.92L6.9 8.35L2 12.75V4.92ZM8.58 9.53L10.43 10.82C10.5974 10.9361 10.7963 10.9984 11 10.9984C11.2037 10.9984 11.4026 10.9361 11.57 10.82L13.42 9.53L18.42 14H3.6L8.58 9.53ZM3 2H19C19.1857 2.00149 19.3674 2.05467 19.5245 2.15358C19.6817 2.25249 19.8083 2.39322 19.89 2.56L11 8.78L2.11 2.56C2.19171 2.39322 2.31826 2.25249 2.47545 2.15358C2.63265 2.05467 2.81428 2.00149 3 2Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
